app/controllers/api/public/v1/providers/locations_controller.rb

```diff
@@ -0,0 +1,31 @@
+module API
+  module Public
+    module V1
+      module Providers
```

Comment:
Why move this to here? Locations can be nested by provider or by course.

Reply:
This is to align the location of the resource in the routes.rb file https://github.com/DFE-Digital/teacher-training-api/blob/fc515bcf6453cb56032827a1d6681d4314ba3bd5/config/routes.rb#L109, as it's nested under the providers module. 

For nested resources, I think it makes sense to have the controllers match the structure of the routes, to keep the concerns separate. The logic for fetching locations for a provider vs a course will be a little different?

Reply:
Agreed, that makes sense. We do need to nest this into courses I believe, though.

Reply:
As discussed, this is worth establishing as a pattern ... so it'd be nice to have an ADR for it.

Reply:
I've done the nesting in https://github.com/DFE-Digital/teacher-training-api/pull/1513/commits/58d486e0d60e4c057cd83c350d0f022375bf358b and also exposed locations for a provider as a separate endpoint https://github.com/DFE-Digital/teacher-training-api/pull/1513/commits/fc231777dbf7ab1c9e05bd7b106a8bd86ce00159